torch/_dynamo/variables/misc.py

```diff
@@ -815,8 +815,12 @@ def call_function(
         from .tensor import NumpyNdarrayVariable
 
         options = VariableTracker.propagate([[self]], [args], [list(kwargs.values())])
-        # lookup method name in torch_np
-        if hasattr(torch_np, self.value.__name__):
+        # lookup method name in torch_np. Things like np.dtype(float) are not supported yet.
+        if self.value.__name__ == "dtype":
+            unimplemented(
+                f"numpy dtype function is not supported yet. Got {self.value}."
+            )
+        elif hasattr(torch_np, self.value.__name__):
```

Comment:
Assert that there is no custom attribute access implementation here. We have utils for this. 

The old code didn't do it, and could potentially cause us to invoke arbitrary user code during the hasattr.

Reply:
`object_has_getattribute` `get_custom_getattr`

Reply:
Found it